server/opts.go

```diff
@@ -3137,6 +3146,8 @@ func ConfigureOptions(fs *flag.FlagSet, args []string, printVersion, printHelp,
 	fs.StringVar(&opts.TLSCaCert, "tlscacert", "", "Client certificate CA for verification.")
 	fs.IntVar(&opts.MaxTracedMsgLen, "max_traced_msg_len", 0, "Maximum printable length for traced messages. 0 for unlimited")
 
+	opts.BasePath = normalizeBasePath(basePath)
```

Comment:
If you think that we really need this option in the command line (config file may be enough), then you would have to move that past the `if err := fs.Parse(args); err != nil {` line. Say line 3176 in this PR.

Also, if we think that this is required in command line, then we need to update main.go with the list of params that can be used (I know, this is not the normal way to do it, but that's how we have been doing so far).

Reply:
True, forgot to change in the `Usage`.

So far I think it's good if you run everything in the command line. I'm pretty new in the NATS world, so I'm not sure if it's more common use config file or command line is also widely used.
Do you have an opinion on that?

Reply:
In very simple cases, you don't need config file and only very few command line params. My problem is starting to have too many command line parameters. I would like to try to keep those to "essentials", not mere convenience. I am sure that that statement does not match existing parameters, but you get the idea.
So if you feel that you need it in your use case and otherwise you would not need a configuration file, maybe that makes sense. If you are going to have a bit of configuration anyway, then I would do without.

Reply:
It makes sense. For my use-case I'll have the config anyway, as it's quite simple to add the command line later I'll remove it for now.